rust_src/src/buffers.rs

```diff
@@ -1843,4 +1847,81 @@ pub fn byte_char_debug_check(b: LispBufferRef, charpos: isize, bytepos: isize) {
     }
 }
 
+// True if B can be used as 'other-than-BUFFER' buffer.
+fn candidate_buffer(b: LispObject, buffer: LispObject) -> bool {
+    match b.as_buffer() {
+        Some(buf) => !b.eq(buffer) && buf.is_live() && !buf.is_hidden(),
+        None => false,
+    }
+}
+
+/// Return most recently selected buffer other than BUFFER.
+/// Buffers not visible in windows are preferred to visible buffers, unless
+/// optional second argument VISIBLE-OK is non-nil.  Ignore the argument
+/// BUFFER unless it denotes a live buffer.  If the optional third argument
+/// FRAME specifies a live frame, then use that frame's buffer list instead
+/// of the selected frame's buffer list.
+///
+/// The buffer is found by scanning the selected or specified frame's buffer
+/// list first, followed by the list of all buffers.  If no other buffer
+/// exists, return the buffer `*scratch*' (creating it if necessary).
+#[lisp_fn(min = "0")]
+pub fn other_buffer(buffer: LispObject, visible_ok: LispObject, frame: LispObject) -> LispObject {
```

Comment:
```suggestion
pub fn other_buffer(buffer: LispObject, visible_ok: LispObject, frame: LispFrameLiveOrSelected) -> LispObject {
```

Followed by:
`let f: LispFrameRef = frame.into();`.

Reply:
Thanks. Done.